### ISSUE TITLE
[FEATURE] Add support to highlight diff/patch

### ISSUE THREAD

**Issue by cryptomilk** [PrivateBin/PrivateBin#189]
Currently if you paste a diff/patch file the language is detected and the the highlighting of the language is used.

You should be able to select Diff and use highlighting for diffs/patch files.

I've looked into implementing it, but prettify.js seems not to support it.

**Comment by elrido** [comment]
The nice thing about prettify is that we don't need to add another UI element to select the language mode and store that as metadata in the paste (forcing support for this on the server side). The downside is that there are a lot of cases like this, where an incorrect highlight mode is auto-selected.

In my personal opinion the benefit of the simplistic model outweighs the downside of making the UI more complicated and having to handle additional metadata on the server side for a special use case.

**Comment by rugk** [comment]
I agree to @elrido, but if you say prettify.js does not support diff highlighting at all you should probably create a PR.

**Comment by cryptomilk** [comment]
I'm sorry but I'm not a web/javascript developer. My skills are somewhere else, so don't expect a patch for that from me.

However from my communities, the pastebin is mostly used to post diffs.

**Comment by kyodev** [comment]
https://github.com/google/code-prettify/issues/62#issuecomment-95294102
this issue was created 8 years ago and tagged enhancement in 2016...
Wait a little longer?  : o /
I can't help you, but I have https://highlightjs.org/ in my notes, automatic,, diff supported, 172 languages but customizable here  https://highlightjs.org/download/ to lighten less than 50 ko.

**Comment by rugk** [comment]
FYI prettify can [handle plugins](https://github.com/google/code-prettify#user-content-for-which-languages-does-it-work), so one could also make a plugin for diffs and if it has a good quality we can naturally include it in PrivateBin.

**Comment by elrido** [comment]
Just a little history: There were forks of zerobin using highlightjs and prettify around. I chose to merge the prettify one into this fork due to it not requiring the user to select anything (which requires a change of the paste format), prettify having line numbering, being more lightweight (including CSS its currently 20 kB) and [back in 2012](https://github.com/isagalaev/highlight.js/graphs/code-frequency) the highlightjs project did not seem very active anymore.

We should certainly at least update prettify, maybe have an option to enable all or selected plugins. A switch to highlightjs would not be opposed, but such a PR would need to be well tested regarding backwards compatibility (= highlightjs needs to have autodetection for pastes with the old format).

**Comment by elrido** [comment]
Note to self: By now highlightJS does offer an automatic detection, so we could switch to it without impact on pastes that have no user selected syntax. The latter could be added as an optional meta data field to the paste, causing no harm in older clients or changes to the format.

**Comment by cryptomilk** [comment]
I think it would make more sense to switch to highlight js. code-prettify just supports a handful of languages and still not support for diff/patch.

**Comment by elrido** [comment]
Yes, thats what I plan to do with this, hence I added it to the project plan for 1.4. :-) Hope it looks ok on the darkstrap theme as well.

**Comment by cryptomilk** [comment]
There are a lot of dark themes available for highlightjs.
